src/topotoolbox/utils.py

```diff
@@ -0,0 +1,98 @@
+"""Provide general utility functions for topotoolbox.
+"""
+import sys
+import os
+
+from urllib.request import urlopen, urlretrieve
+
+from .grid_object import GridObject
+
+__all__ = ["load_dem", "get_dem_names", "read_tif"]
+
+DEM_SOURCE = "https://raw.githubusercontent.com/TopoToolbox/DEMs/master"
+DEM_NAMES = f"{DEM_SOURCE}/dem_names.txt"
+
+
+def read_tif(path: str) -> GridObject:
+    """Generate a new GridObject from a .tif file.
+
+    Args:
+        path (str): path to .tif file.
+
+    Returns:
+        GridObject: A new GridObject of the .tif file.
+    """
+    return GridObject(path)
+
+
+def get_dem_names() -> list[str]:
+    """Returns a list of provided example Digital Elevation Models (DEMs).
+    Requires internet connection to download available names.
+
+    Returns:
+        list[str]: A list of strings, where each string is the name of a DEM.
+    """
+    with urlopen(DEM_NAMES) as dem_names:
+        dem_names = dem_names.read().decode()
+
+    return dem_names.splitlines()
+
+
+def load_dem(dem: str, cache=True) -> GridObject:
+    """Downloads DEM from wschwanghart/DEMs repository. 
+    Find possible names by using 'get_dem_names()'
+
+    Args:
+        dem (str): Name of dem about to be downloaded
+        cache (bool, optional): If true the dem will be cached. 
+        Defaults to True.
+
+    Returns:
+        GridObject: A GridObject generated from the downloaded dem.
+    """
+
+    url = f"{DEM_SOURCE}/{dem}.tif"
+
+    if cache:
+        cache_path = os.path.join(get_save_location(), f"{dem}.tif")
+
+        if not os.path.exists(cache_path):
+            urlretrieve(url, cache_path)
+
+        full_path = cache_path
```

Comment:
Can we provide a `clear_cache` function or something similar to allow users to empty this directory if they decide they don't need the data anymore?